packages/lib-core/src/types/k8s.ts

```diff
@@ -0,0 +1,122 @@
+export type K8sVerb =
+  | 'create'
+  | 'get'
+  | 'list'
+  | 'update'
+  | 'patch'
+  | 'delete'
+  | 'deletecollection'
+  | 'watch';
+
+export enum Operator {
+  Exists = 'Exists',
+  DoesNotExist = 'DoesNotExist',
+  In = 'In',
+  NotIn = 'NotIn',
+  Equals = 'Equals',
+  NotEqual = 'NotEqual',
+  GreaterThan = 'GreaterThan',
+  LessThan = 'LessThan',
+  NotEquals = 'NotEquals',
+}
+
+export type MatchExpression = {
+  key: string;
+  operator: Operator | string;
+  values?: string[];
+  value?: string;
+};
+
+export type MatchLabels = {
+  [key: string]: string;
+};
+
+export type Selector = {
+  matchLabels?: MatchLabels;
+  matchExpressions?: MatchExpression[];
+};
+
+enum BadgeType {
+  DEV = 'Dev Preview',
+  TECH = 'Tech Preview',
+}
+
+export type K8sModel = {
```

Comment:
This is going to be covered & expanded on eventually from Vojtech's PR: https://github.com/openshift/dynamic-plugin-sdk/pull/30/files#diff-01541dad1d9bd6352f582e703d8f5a8e5c2b81b6b92e7cab86fd07d542c5d8b6R7

And this has a lot of weird OpenShift Console use-cases we are not yet ready or ever want to support (ie the badge prop)